core/src/main/kotlin/io/customer/sdk/data/store/GlobalPreferenceStore.kt

```diff
@@ -10,7 +10,11 @@ import androidx.core.content.edit
 interface GlobalPreferenceStore {
     fun saveDeviceToken(token: String)
     fun getDeviceToken(): String?
-
+    fun saveUserId(token: String)
```

Comment:
I think we shouldn't be using global store for in-app purposes. This can be greatly misleading and can pose challenges to anyone reading the codebase in future. I think we should either have a consistent global store managed by module identifying and keeping track of profiles (i.e. `datapipelines`). or create separate store for in-app to isolate it from other modules.

Reply:
> I think we should either have a consistent global store managed by module identifying and keeping track of profiles (i.e. datapipelines)

These modules are independent and should not be bound to `datapipelines`, the idea of adding it in global was if the value is present there, no matter which source added it, other modules can utilize it too, if the value isn't there, they can rely on their own local values. 

I am okay with creating module specific preference store, but we would need to do it for every module, if that's okay, happy to create one.

Reply:
Discussed on call, we'll add module specific preference store to in-app module